apiclient.js

```diff
@@ -3043,20 +3056,80 @@ class ApiClient {
         return this.getJSON(url);
     }
 
+    getVideoCodecs(userId, options) {
+
+        if (!userId) {
+            throw new Error("null userId");
+        }
+
+        options = options || {};
+        options.userId = userId;
+
+        const url = this.getUrl("VideoCodecs", options);
+
+        return this.getJSON(url);
+    }
+
+    getAudioCodecs(userId, options) {
+
+        if (!userId) {
+            throw new Error("null userId");
+        }
+
+        options = options || {};
+        options.userId = userId;
+
+        const url = this.getUrl("AudioCodecs", options);
+
+        return this.getJSON(url);
+    }
+
+    getSubtitleCodecs(userId, options) {
+
+        if (!userId) {
+            throw new Error("null userId");
+        }
+
+        options = options || {};
+        options.userId = userId;
+
+        const url = this.getUrl("SubtitleCodecs", options);
+
+        return this.getJSON(url);
+    }
+
     getPrefixes(userId, options) {
 
         if (!userId) {
             throw new Error("null userId");
         }
 
-        if (!this.isMinServerVersion('3.6.0.60')) {
+        if (!this.isMinServerVersion('3.6.0.85')) {
```

Comment:
Looks like some API calls were removed and they just returned hardcoded lists for 3.6+?

We should probably remove any of these checks when we do some cleanup.

Reply:
Why introduce stuff that is easily cleanable to be cleaned up later? I mean, why lay it off if it could be more-or-less easily be done now?

Reply:
This PR is just dealing with bringing in upstream changes not fixing issues with the codebase. I would be fine merging it and then moving on to cleaning up unused files and the like.

Reply:
Jellyfin will still work with this since 10.0.0 > 3.6.0.85 :)

Reply:
I was limiting this PR to pulling in the upstream changes since we will have _a lot_ of cleanup to do, but I don't mind addressing these specific items if they are a hangup.